File: LKT/PROGRAMMERS/Lv.3/단어변환/Solution.java
class Solution {
    static boolean[] visited;
    static int answer;

    public int solution(String begin, String target, String[] words) {
        visited = new boolean[words.length];
        dfs(begin, target, words, 0);
        return answer;
    }

    static void dfs(String begin, String target, String[] words, int cnt) {
        if (begin.equals(target)) {
            answer = cnt;
            return;
        }

        for (int i = 0; i < words.length; i++) {
            if (!visited[i]) {
                if (check(begin, words[i])) {
                    visited[i] = true;
                    dfs(words[i], target, words, cnt + 1);
                    visited[i] = false;
                }
            }
        }
    }

    static boolean check(String s1, String s2) {
        int cnt = 0;
        for (int i = 0; i < s1.length(); i++) {
            if (cnt > 1)
                break;
            if (s1.charAt(i) != s2.charAt(i))
                cnt++;
        }

        return cnt == 1;
    }
}

File: LKT/PROGRAMMERS/Lv.3/단어변환/solution.cpp
#include <iostream>
#include <string>
#include <vector>
#define cstr(x) x.c_str()

using namespace std;

vector<vector<int>> v;

int strcmp(const char *s1, const char *s2) {
    int res = 0;
    while(*s1 != '\0' && *s2 != '\0') {
        if (*s1 != *s2) res++;
        s1++;
        s2++;
    }

    return res;
}

int solution(string begin, string target, vector<string> words) {
    

    return answer;
}

int main() {
    string begin = "hit";
    string target = "cog";  
    vector<string> words[2] = {{"hot", "dot", "dog", "lot", "log", "cog"}, {"hot", "dot", "dog", "lot", "log"}};

    for (int i = 0; i < 2; i++) {
        cout << solution(begin, target, words[i]) << "\n";
    }
}